### PR TITLE
List of speakers slide

### PR DESCRIPTION
**Pull request by FinnStutzenstein**

### CODE REVIEW

File: client/src/app/slides/common-list-of-speakers/common-list-of-speakers-slide.component.html [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
         <h1>
             {{ 'List of speakers' | translate }}
             <mat-icon *ngIf="data.data.closed">lock</mat-icon>
         </h1>
         <h2>
-            {{ getTitle() }}
-            <span *ngIf="getSpeakersCount() > 0 && data.data.list_of_speakers_show_amount_of_speakers_on_slide">
-                &ndash; {{ getSpeakersCount() }} <span>{{ 'Speakers' | translate }}</span>
+            {{ title }}
+            <span *ngIf="this.data.data.number_of_last_speakers !== null && this.data.data.number_of_last_speakers !== undefined">
```
Comment (tsiegleauq) [inline client/src/app/slides/common-list-of-speakers/common-list-of-speakers-slide.component.html:9]: `ngIf="Number.isFinite(data.data?.number_of_last_speakers)"`

File: client/src/app/slides/common-list-of-speakers/common-list-of-speakers-slide.component.html [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
+                &ndash; {{ this.data.data.number_of_last_speakers }} <span>{{ 'Speakers' | translate }}</span>
```
Comment (tsiegleauq) [inline client/src/app/slides/common-list-of-speakers/common-list-of-speakers-slide.component.html:10]: used 2nd time, make a getter

File: client/src/app/slides/common-list-of-speakers/common-list-of-speakers-slide.component.html [hunk 3]
```diff
@@ -16,26 +16,32 @@ <h2>
...
     <!-- Current speaker -->
     <div *ngIf="data.data.current" class="currentSpeaker">
         <mat-icon class="micicon">mic</mat-icon>
         {{ data.data.current.user }}
-        <mat-icon *ngIf="data.data.current.marked">star</mat-icon>
-        <mat-icon *ngIf="data.data.current.point_of_order" color="warn"> warning</mat-icon>
+        <mat-icon *ngIf="data.data.current.speech_state === SpeechState.CONTRIBUTION">star</mat-icon>
+        <mat-icon *ngIf="data.data.current.speech_state === SpeechState.PRO" class="green-text">add_circle</mat-icon>
+        <mat-icon *ngIf="data.data.current.speech_state === SpeechState.CONTRA" class="red-warning-text">remove_circle</mat-icon>
+        <mat-icon *ngIf="data.data.current.point_of_order" color="warn">warning</mat-icon>
```
Comment (tsiegleauq) [inline client/src/app/slides/common-list-of-speakers/common-list-of-speakers-slide.component.html:33]: duplicate from above, use ngTemplate to encapsule duplications

File: client/src/app/slides/common-list-of-speakers/common-list-of-speakers-slide.component.html [hunk 4]
```diff
@@ -16,26 +16,32 @@ <h2>
...
     <div *ngIf="data.data.waiting && data.data.waiting.length">
         <ol class="nextSpeakers">
             <li *ngFor="let speaker of data.data.waiting">
                 {{ speaker.user }}
-                <mat-icon *ngIf="speaker.marked">star</mat-icon>
-                <mat-icon *ngIf="speaker.point_of_order" color="warn"> warning</mat-icon>
+                <mat-icon *ngIf="speaker.speech_state === SpeechState.CONTRIBUTION">star</mat-icon>
+                <mat-icon *ngIf="speaker.speech_state === SpeechState.PRO" class="green-text">add_circle</mat-icon>
+                <mat-icon *ngIf="speaker.speech_state === SpeechState.CONTRA" class="red-warning-text">remove_circle</mat-icon>
+                <mat-icon *ngIf="speaker.point_of_order" color="warn">warning</mat-icon>
```
Comment (tsiegleauq) [inline client/src/app/slides/common-list-of-speakers/common-list-of-speakers-slide.component.html:44]: another duplicate

File: client/src/app/slides/common-list-of-speakers/common-list-of-speakers-slide.component.html [hunk 5]
```diff
@@ -16,26 +16,32 @@ <h2>
     <div *ngIf="data.data.finished && data.data.finished.length" class="lastSpeakers">
         <div *ngFor="let speaker of data.data.finished">
             {{ speaker.user }}
-            <mat-icon *ngIf="speaker.marked">star</mat-icon>
-            <mat-icon *ngIf="speaker.point_of_order" color="warn"> warning</mat-icon>
+            <mat-icon *ngIf="speaker.speech_state === SpeechState.CONTRIBUTION">star</mat-icon>
+            <mat-icon *ngIf="speaker.speech_state === SpeechState.PRO" class="green-text">add_circle</mat-icon>
+            <mat-icon *ngIf="speaker.speech_state === SpeechState.CONTRA" class="red-warning-text">remove_circle</mat-icon>
+            <mat-icon *ngIf="speaker.point_of_order" color="warn">warning</mat-icon>
```
Comment (tsiegleauq) [inline client/src/app/slides/common-list-of-speakers/common-list-of-speakers-slide.component.html:22]: first definition

**Review by tsiegleauq — APPROVED** [review]
Works fine. Template code could be cleaner.

**Comment by FinnStutzenstein** [comment]
TODO: `number_of_last_speakers` -> `number_of_waiting_speakers`

File: client/src/app/shared/components/projector/projector.component.ts [hunk 6]
```diff
@@ -168,10 +168,14 @@ export class ProjectorComponent extends BaseComponent implements OnDestroy {
             this.offlineBroadcastService.isOfflineObservable.subscribe(isOffline => (this.isOffline = isOffline))
         );
 
-        this.slides = this.projectorSubject.pipe(
-            switchMap(projector => projector?.current_projections_as_observable || []),
-            map(projections =>
-                projections.map(
+        const trigger$ = merge(
```
Comment (FinnStutzenstein) [inline client/src/app/shared/components/projector/projector.component.ts:171]: These changes have to be done to the presentation control component, too.

@tsiegleauq You can merge this PR if you want, but this is still to be done. I can't do it until next week

Comment (tsiegleauq) [reply client/src/app/shared/components/projector/projector.component.ts:171]: i'll amend it tomorrow

File: client/src/app/site/projector/components/presentation-control/presentation-control.component.ts [hunk 7]
```diff
@@ -21,27 +21,16 @@ import { ViewProjector } from '../../models/view-projector';
 export class PresentationControlComponent extends BaseComponent {
     @Input()
     public set projector(projector: ViewProjector) {
-        this.projectorSubject.next(projector);
+        this.projections = projector.current_projections_as_observable;
```
Comment (tsiegleauq) [inline client/src/app/site/projector/components/presentation-control/presentation-control.component.ts:24]: @FinnStutzenstein seems it was not even required.

**Pull request merged**